drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#define DT_DRV_COMPAT nxp_irtc
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/irq.h>
+
+struct nxp_irtc_config {
+	RTC_Type *base;
+	uint8_t alarm_match_flag;
+	uint8_t clock_select;
+	bool is_output_clock_enabled;
+	void (*irq_config_func)(const struct device *dev);
+};
+
+struct nxp_irtc_data {
+	bool is_dst_enabled;
+#ifdef CONFIG_RTC_ALARM
+	rtc_alarm_callback alarm_callback;
+	void *alarm_user_data;
+	uint16_t *alarm_mask;
+	bool is_alarm_pending;
+#endif /* CONFIG_RTC_ALARM */
+#ifdef CONFIG_RTC_UPDATE
+	rtc_update_callback update_callback;
+	void *update_user_data;
+#endif /* CONFIG_RTC_ALARM */
+#ifdef CONFIG_RTC_CALIBRATION
+	int32_t calibration;
+#endif /* CONFIG_RTC_CALIBRATION */
+};
+
+/* The IRTC Offset is 2112 instead of 1990 [2112 - 1990] -> [212] */
+#define RTC_NXP_IRTC_YEAR_OFFSET 212
+
+/* Utility function for testing if bit for alarm mask is enabled. */
+#define RTC_NXP_RTCTIME_COMPARE_FLAG(mask, flag) (((mask) == 0) || (((mask) & (flag)) && (flag)))
```

Comment:
can't you use util_macro.h?

Reply:
What macro are you thinking of here?

Reply:
FIELD_GET?

Reply:
These macros do only parts if that of what I want from my macro, it would seem fitting that I just use a macro from scratch.

Reply:
look in eth_nxp_enet_qos.h, should we do something generic for NXP drivers to use SDK defines like this
(non blocking or even suggestion for this PR just a question)